pkg/shared/utils/utils.go

```diff
@@ -53,6 +57,39 @@ func RunCommand(cmd *exec.Cmd) ([]byte, error) {
 	return outb.Bytes(), nil
 }
 
+func RunCommandAndParseOutputLineByLine(cmd *exec.Cmd, pfn processFn) error {
+	// Get a pipe to read from standard out
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stdout pipe: %w", err)
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("failed to get stderr pipe: %w", err)
+	}
+
+	// Start the command and check for errors
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("failed to start command: %w", err)
+	}
+
+	// Merge stdout and stderr
+	merged := io.MultiReader(stderr, stdout)
```

Comment:
do we want to pass stdout and stderr to the process function without knowing which is which? It seems like we might get a bunch of unexpected log/info lines coming through which the process function will have to omit. It might not be as big of a problem for Yara, but I know other tools output a log of info stuff on stderr.

It might be better to have a separate scanner go routine, and just throw the stderr lines straight into a logger.Debug so that they appear in the VMClarity scanner output if you turn on debug mode so that we can see them, but we don't need to cache them in memory or pass them to the process function.

Reply:
I've added a simple stderr logging for now.

Reply:
Love it 👍